### PR TITLE
RavenDB-19563 - Index mismatch after snapshot restore

### PR DESCRIPTION
**Pull request by grisha-kotler**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-19563/Index-mismatch-after-snapshot-restore

### Additional description

Change the order in which we perform the snapshot backup - first the indexes then the documents.

### Type of change

- Bug fix
- 
### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### CODE REVIEW

**Comment by ppekrol** [comment]
@grisha-kotler can you check failing tests?

File: src/Raven.Server/Documents/TombstoneCleaner.cs [hunk 1]
```diff
@@ -64,6 +66,13 @@ public void Unsubscribe(ITombstoneAware subscription)
             }
         }
 
+        public IDisposable PreventTombstoneCleaning(long minTombstoneToDelete)
+        {
+            _minTombstoneToDelete = minTombstoneToDelete;
+
+            return new DisposableAction(() => { _minTombstoneToDelete = long.MaxValue; });
```
Comment (ppekrol) [inline src/Raven.Server/Documents/TombstoneCleaner.cs]: it's counter-intuitive
_minTombstoneToDelete = Max => do not delete anything IMO

Make it nullable and set it to null?

File: src/Raven.Server/Documents/TombstoneCleaner.cs [hunk 2]
```diff
@@ -178,6 +187,8 @@ internal TombstonesState GetState(bool addInfoForDebug = false)
                 _subscriptionsLocker.Release();
             }
 
+            result.MinAllDocsEtag = Math.Min(result.MinAllDocsEtag, _minTombstoneToDelete);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/TombstoneCleaner.cs]:
```
var minTombstoneToDelete = _minTombstoneToDelete;
if (minTombstoneToDelete.HasValue)
    result.MinAllDocsEtag = Math.Min(result.MinAllDocsEtag, minTombstoneToDelete.Value);
```

Comment (grisha-kotler) [reply src/Raven.Server/Documents/TombstoneCleaner.cs]: done

File: src/Raven.Server/Documents/TombstoneCleaner.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Server/Documents/TombstoneCleaner.cs]: done

**Comment by ppekrol** [comment]
@grisha-kotler can you check failing tests?

**Pull request merged**